fairlearn/preprocessing/_optimPreproc.py

```diff
@@ -0,0 +1,213 @@
+import numpy as np
+import pandas as pd
+from ._optimPreproc_helper import DTools
+
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class OptimizedPreprocessor:
+    def __init__(self, optimizer, optim_options, verbose=False, seed=None) -> None:
+        super().__init__()
+
+        self.seed = seed
+        self.optimizer = optimizer
+        self.optim_options = optim_options
+        self.verbose = verbose
+
+    def fit(self, df, X_features, Y_features, D_features):
```

Comment:
I can use a similar approach and it should not take much time to modify.

There are two ways Correlation remover is working right now-
1) Using numpy arrays, since this does not have feature names, index has to be passed
2)When a pandas dataframe is passed, it can take advantage of column names

Both are being passed using `sensitive_feature_ids`

Reply:
Apologies for the delayed response since I had some interviews! 

I changed the code according to scikit documentation - https://scikitlearn.org/stable/modules/generated/sklearn.base.TransformerMixin.html

Please review @adrinjalali